x/authz/keeper/msg_server.go

```diff
@@ -43,6 +43,9 @@ func (k Keeper) Grant(ctx context.Context, msg *authz.MsgGrant) (*authz.MsgGrant
 	if err := k.MsgRouterService.CanInvoke(ctx, t); err != nil {
 		return nil, sdkerrors.ErrInvalidType.Wrapf("%s doesn't exist", t)
 	}
+	if t == sdk.MsgTypeURL(&authz.MsgGrant{}) {
```

Comment:
Maybe let's add a comment on why it is done here (msg server) and not in the keeper?
My guess is you did it like this so that modules can still create a grant of grants if needed, and this only break the user flow and not module code. If so I would precise that so it doesn't get refactored away in the futurre.

Reply:
added the docs